### ISSUE TITLE
FHS

### ISSUE THREAD

**Issue by jubalh** [CoatiSoftware/Sourcetrail#808]
It would be good if Sourcetrail would follow [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/index.html).

**Comment by egraether** [comment]
Our Linux build is a bit messy at the moment, mostly because none of us is a real Linux expert. Feel free to restructure.

**Comment by fpesari** [comment]
This is very important if the goal is to make Sourcetrail popular, and more users means more contributors, which in turn means more languages supported :smile_cat: 

[This post](https://github.com/CoatiSoftware/Sourcetrail/issues/758#issuecomment-559032620) by @jubalh basically summarized the things most needed to easily package Sourcetrail, which are:
1. FHS support, which is very trivial actually - more or less, the executable goes in `/usr/bin`, libraries go in `/usr/lib` and data goes in `/usr/share/Sourcetrail`. The only tricky thing is that some distros (like openSUSE) use `/usr/lib64` instead of `/usr/lib` when building on 64-bit systems, but CMake is able to take care of that. Then there are a bunch of things like icons and manuals which packagers can install themselves, if that's a problem - the main priority is binaries, libraries and data used by the program.
2. Use system libraries
3. (This is mine) Do not require an internet connection to build at any point. This is very important for automatic build systems, which are used to distribute official packages

**Comment by jubalh** [comment]
> Do not require an internet connection to build at any point. This is very important for automatic build systems, which are used to distribute official packages

I don't think it does that. We build Sourcetrail for openSUSE in a build system with no network.

**Comment by mlangkabel** [comment]
Currently Sourcetrail requires an internet connection when building for the first time to
* let Maven download the Java dependencies for building the java indexer
* download the configured release of the Sourcetrail Python indexer for Python language support.

**Comment by jubalh** [comment]
Does it require this to build? Because the latest two releases build just fine in our network-less build system.

**Comment by mlangkabel** [comment]
It is not required if you build Sourcetrail without Python and Java language support (can be enabled or disabled in cmake).

**Comment by jubalh** [comment]
Right. I didn't do that.

Then I agree that this should be changed :)

**Comment by PyroSamurai** [comment]
I suggest you familiarize yourself with [these](https://specifications.freedesktop.org/) if you are going to deal with Linux-based desktops and not GUI-less terminals. Especially the basedir-spec. FHS is hardly the full story for "desktop" applications.

**Comment by egraether** [comment]
With #1034 we unified all of our Linux release packages to depend on [linuxdeployqt](https://github.com/probonopd/linuxdeployqt).

Now the [setup/Linux/createPackages.sh](https://github.com/CoatiSoftware/Sourcetrail/blob/9a8fa94905f021a8d0da15c312752826873068e6/setup/Linux/createPackages.sh) script creates both Linux release packages (`.tar.gz` and `.AppImage`) and documents everything that goes into each package.

I think this is as far as I can take it with my Linux knowledge. Anyone looking to improve this, can start working from there.
